src/nccl_ofi_rdma.c

```diff
@@ -375,21 +340,23 @@ static int write_topo_file(nccl_ofi_topo_t *topo)
 		NCCL_OFI_WARN("Unable to set NCCL_TOPO_FILE.ERROR: %s",
 			      strerror(errno));
 		ret = -errno;
-		goto unlock;
+		goto error;
 	}
 
-	rc = atexit(unlink_topo_file);
-	if (rc != 0) {
-		NCCL_OFI_WARN("Failed to set exit function");
-		ret = -1;
-		goto exit;
+	goto exit;
 
+error:
+	if (file) {
+		fclose(file);
+		/* fclose will close the file descriptor, so avoid closing it
+		   again below. */
+		topo_fd = -1;
```

Comment:
This code is overly complicated because you're avoiding a return in the case that fdopen() fails.  Just fix that and then you don't need all this nonsense about setting topo_fd to -1.